### PR TITLE
Release 4.1.0 | Aesthetic, functionality changes and a rewrite

### LINKED ISSUE #117: [Implemented soonTM] New package.json
As we're aproaching v4.1, I'd say we should get a new package.json. Theres things that don't need to be there (like bufferutil) and things that work just fine in newer versions.

~I've been currently working on a "new" one, and I'll Pr it and wait for opinions~

- [x] Remove bufferutil
- [x] Remove bluebird
- [x] Add String (`"string": "*"`)
- [x] Change random-puppy to random-animal
- [x] ~~Replace "money"~~ Fix money

### LINKED ISSUE #120: [Formatting] Fix all them formatting issues that I caused
Commands and such have poor formatting, mainly due to IntelliJ.

As such, for v4.1 we'll have everything formatted nice and pretty. Maybe even add docs cause why not

### PR DESCRIPTION
**Pull request by vladfrangu**
## Things to consider

This version changes **everything**. You will have to update your values from `config.json` and `auth.json` to their respective JS counterparts.

You will also notice 1 pre-assigned API key in `auth.js`, and thats no mistake. 

## Warning

Do **NOT** Run this branch. Don't. At the time of writing this (or rewriting), the bot has **NO** core features available. Its a full rewrite. Wait till I'll edit this and tell you you *may* go ahead and give it a test run, at your own risk.
And, if you find any bugs, or you'd like a code refactor, drop a comment.

## The aim of this Pull Request

This Pull Request is aimed to **rewrite** and improve the bot. Everything will look better, should work faster, and best of all, everything should and will work.

## What does this Pull Request Contain

- A full bot rewrite, to Discord.js

- Improvements to the command looks and feel

- Addition of smaller things

- A new god damn Extension API

## Current Progress

- [x] Delete the old files

- [x] Rewrite the main base, using `async`

- [ ] Port all of the commands

- [x] Fix broken shit (`giveaway`, `lottery`, `MOTD constantly repeating` etc)

- [ ] Test things

- [x] Pray to Vlag

## Old Credit should go where its due

(This is here for consistency, these credits affect the old Eris commits)

Thanks TTtie ( @TTtie ) for the help provided with the `GetUserWithoutREST`, helping me test the `info` command due to me refactoring some code and with the inspiration for the `debug` command

### CODE REVIEW

**Comment by KingDGrizzle** [comment]
Mark my words @GilbertGobbels if you don't do a full check on all the code and commands I'll kill you!

**Comment by TTtie** [comment]
~~and i raprot u to dismcord for telling u hurt gilbert~~

**Comment by KingDGrizzle** [comment]
*cough* image

**Comment by TTtie** [comment]
O kurva.....

**Comment by KingDGrizzle** [comment]
Hey @GilbertGobbels promise me that you will respect the eslint rules or I'll personally setup travis and tests for eslint which will prevent PR's from merging unless rules are met:tm:

~Don't do a javascript await meme i swear~

**Comment by KingDGrizzle** [comment]
Wasn't eslint, i just did it cause

> It works

> It makes more sense

**Comment by vladfrangu** [comment]
Need to rewrite the main message.. fuck

File: Discord.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+const reload = require("require-reload")(require);
+const { RankScoreCalculator: computeRankScores, ModLog } = require("./Modules/");
+
+const privateCommandModules = {};
+const commandModules = {};
+
+/* eslint-disable max-len */
+module.exports = (db, configJS, configJSON) => {
+	// Create a Discord.js Bot Client
+	const Discord = require("discord.js");
```
Comment (TTtie) [inline Discord.js]: rip eris in gab 2016(?) - 2017.

**Comment by vladfrangu** [comment]
10/10 that was either a sarcastic thanks, or a legit thanks 🤣

**Comment by vladfrangu** [comment]
When codacity will bang us 😂

**Comment by TTtie** [comment]
DAhaa

**Comment by vladfrangu** [comment]
Version 4.1 it is! 🎉

**Comment by TTtie** [comment]
:eyes: 

screenshot_20170803-124100

**Comment by vladfrangu** [comment]
Mark this, as both Travis and Codacy gave their checks!

**Comment by TTtie** [comment]
👌

**Comment by vladfrangu** [comment]
So many freakin checks!

**Comment by TTtie** [comment]
god bless

File: Web/views/pages/landing.ejs [hunk 2]
```diff
@@ -607,13 +518,13 @@
...
 											<p class="title is-3">King - Vlad</p>
-											<p class="subtitle is-5">Developer, Embedder</p>
+											<p class="subtitle is-5">Developer, Support</p>
 										</div>
 									</div>
 
 									<div class="content">
-										I do embeds <span class="icon">
-											<i class="fa fa-id-card-u"></i>
+										Command Wizzard <span class="icon">
```
Comment (vitalkanev) [inline Web/views/pages/landing.ejs]: typo here

Comment (vladfrangu) [reply Web/views/pages/landing.ejs]: Its `Wizard` isn't it? 😄

Comment (vitalkanev) [reply Web/views/pages/landing.ejs]: yes it is...

**Comment by oskikiboy** [comment]
Using figlet module for customisable ascii art

**Comment by vladfrangu** [comment]
Except we probably don't want that customized so uh.. @GilbertGobbels opinions?

**Comment by vitalkanev** [comment]
Can you just fix Travis...? I'm pretty confused

**Comment by vladfrangu** [comment]
Its the database being down, and thats @GilbertGobbels DB so.. It will get fixed _soon_

**Comment by TTtie** [comment]
:tada: TRAVIS WORKS :eyes:

**Comment by vitalkanev** [comment]
@TTTie I think there is no "bot run" - only eslint

**Comment by jtsshieh** [comment]
Isn't Eris faster than discord.js? Eris doesn't have all the stupid bloaty functions (tag) that discord.js has.

**Comment by vladfrangu** [comment]
I'm genuinely confused.. Unless you can explain why that is, I'm remaining confused.

> Stupid Bloaty Functions

What?

**Comment by jtsshieh** [comment]
Discord.js has a bunch of functions to **Make your life easier** but not necessarily making the bot faster. When aiming for speed, usually Eris is the winner because it doesn't have all the functions that **Make your life easier** but still it maintains the basic functions and is very optimized for speed. So, your title is kinda misleading.

**Comment by vladfrangu** [comment]
How the? What?!

Just cause it has more functions, it doesn't mean its `slower`.
Thanks to Discord.js, we get a lot of features that are **needed** for our use internally, as well as allows us to easily multi-thread this bot.

Just adding more features to `make your life easier` doesn't mean its more `bloated`. Just use what you want.

**Comment by TTtie** [comment]
define the features

**Comment by SunburntRock89** [comment]
This pains me http://gawesomebot.works.wtf/how/5d244.png

**Comment by SunburntRock89** [comment]
I regret ever commenting because now I'm subscribed lol

**Comment by wasdennnoch** [comment]
I mean, you can just unsubscribe via the button in the right sidebar. Or unwatch the project at the top of the page.

**Comment by vitalkanev** [comment]
I guess done @GilbertGobbels?

**Comment by vladfrangu** [comment]
Please note that this can be counted as `PR 1 out of many` when it comes to the rewrite. Expect more PRs

**Comment by TTtie** [comment]
that reminds me of the thing that I should work on #202

**Pull request merged**